pkg/resource/deploy/step_generator.go

```diff
@@ -399,75 +402,182 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, *re
 	return []Step{NewCreateStep(sg.plan, event, new)}, nil
 }
 
-func (sg *stepGenerator) GenerateDeletes() []Step {
+func (sg *stepGenerator) GenerateDeletes() []antichain {
```

Comment:
It feels like it may be a bit cleaner to let this method generate the list of steps and then let the caller schedule that list (rather than splitting the responsibility for step generation across multiple methods). That would eliminate a fair bit of the conditional code here, and would allow `scheduleDeletes` to omit the `pendingDeletesAreReplaces` parameter.

Reply:
To make this more concrete, I would change the signature of `scheduleDeletes` s.t. it takes a `[]Step`. It can then decide whether to just return the slice as-is (in the case that dependency information is not conservatively correct) or to schedule using antichain decomposition.

Reply:
Makes sense - will do.